### PR TITLE
FIX: Back button action and position on glimmer topic timeline

### PR DESCRIPTION
**Pull request by janzenisaac**
This PR introduces a proper `action` to the topic timeline `back-button` which will fix the button not being clickable (or functional) as well as removing a duplicate setting of `this.lastReadTop` which was causing odd positionings of the button.

This is very difficult to test due to the fact you have to manage the "read history" for a user to have the back button populate. We will have to move forward without one (as we did in the last version of the timeline 😅) for now.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/topic-timeline/back-button.hbs [hunk 1]
```diff
@@ -2,7 +2,7 @@
   @type="button"
   @class="btn-primary btn-small back-button"
   @title={{i18n "topic.timeline.back_description"}}
-  @onClick={{@onClick}}
+  @action={{@goBack}}
 >
   {{i18n "topic.timeline.back"}}
-</DButton>
\ No newline at end of file
+</DButton>
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/topic-timeline/back-button.hbs:8]: Why does GH UI show there are changes on the last lines of those hbs files? 🤨

File: app/assets/javascripts/discourse/app/components/topic-timeline/container.hbs [hunk 2]
```diff
@@ -96,11 +96,11 @@
         onClick={{this.updatePercentage}}
       ></div>
 
-      {{#if this.hasBackPosition}}
+      {{#if (and this.hasBackPosition this.showButton)}}
         <div class="timeline-last-read" style={{this.lastReadStyle}}>
           {{d-icon "minus" class="progress"}}
           {{#if this.showButton}}
-            <TopicTimeline::BackButton @onClick={{this.goBack}} />
+            <TopicTimeline::BackButton @goBack={{this.goBack}} />
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/topic-timeline/container.hbs:103]: How about renaming that arg to `onGoBack` or similar? i.e. to indicate it's a function/action/callback

File: app/assets/javascripts/discourse/app/components/topic-timeline/back-button.hbs (hunk 1 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/topic-timeline/back-button.hbs:8]: @janzenisaac you likely need to update your editor config to stop it adding newlines to the end of `.hbs` files - it's a prettier violation. (which was previously being ignored by GitHub actions - fixed by https://github.com/discourse/discourse/pull/20139)

File: app/assets/javascripts/discourse/app/components/topic-timeline/container.hbs (hunk 2 above)
Comment (janzenisaac) [reply app/assets/javascripts/discourse/app/components/topic-timeline/container.hbs]: Added in last commit 👍

**Review by CvX — APPROVED** [review]

**Pull request merged**